src/SBInterpolatedImage.cpp

```diff
@@ -895,31 +881,61 @@ namespace galsim {
         }
         _flux = kValue(Position<double>(0.,0.)).real();
         dbg<<"flux = "<<_flux<<std::endl;
+        setCentroid();
+    }
+
+    void SBInterpolatedKImage::SBInterpolatedKImageImpl::setCentroid() const {
```

Comment:
I had some fun with a profiler visualization tool (https://github.com/jrfonseca/gprof2dot) I saw someone using on slack.  Here's the output for the test_CRG_noise.py devel script:

![output](https://user-images.githubusercontent.com/3650485/27196387-185bd4f4-51bf-11e7-823f-641702d1562b.png)

The InterpolatedKImage initializer (the box on the left labeled `interpolatedimage.py:677:__init__`) takes 2.5% of the total run time, of which 1.7% is the Hermitian conjugacy sanity check (the `numeric:2310:allclose` box).  So I think the centroid calculation is probably sufficiently fast.